src/mbgl/style/sources/tile_source.cpp

```diff
@@ -0,0 +1,108 @@
+#include <mbgl/storage/file_source.hpp>
+#include <mbgl/style/conversion/json.hpp>
+#include <mbgl/style/conversion/tileset.hpp>
+#include <mbgl/style/layer.hpp>
+#include <mbgl/style/source_observer.hpp>
+#include <mbgl/style/sources/tile_source.hpp>
+#include <mbgl/style/sources/tile_source_impl.hpp>
+#include <mbgl/tile/tile.hpp>
+#include <mbgl/util/async_request.hpp>
+#include <mbgl/util/exception.hpp>
+#include <mbgl/util/mapbox.hpp>
+
+namespace mbgl {
+namespace style {
+
+TileSource::TileSource(std::string id,
+                           variant<std::string, Tileset> urlOrTileset_,
+                           uint16_t tileSize,
+                           SourceType sourceType)
+    : Source(makeMutable<Impl>(sourceType, std::move(id), tileSize)),
+      urlOrTileset(std::move(urlOrTileset_)) {}
+
+TileSource::~TileSource() = default;
+
+const TileSource::Impl& TileSource::impl() const {
+    return static_cast<const Impl&>(*baseImpl);
+}
+
+const variant<std::string, Tileset>& TileSource::getURLOrTileset() const {
+    return urlOrTileset;
+}
+
+std::optional<std::string> TileSource::getURL() const {
+    if (urlOrTileset.is<Tileset>()) {
+        return {};
+    }
+
+    return urlOrTileset.get<std::string>();
+}
+
+uint16_t TileSource::getTileSize() const {
+    return impl().getTileSize();
+}
+
+void TileSource::loadDescription(FileSource& fileSource) {
+    if (urlOrTileset.is<Tileset>()) {
+        baseImpl = makeMutable<Impl>(impl(), urlOrTileset.get<Tileset>());
+        loaded = true;
+        observer->onSourceLoaded(*this);
+        return;
+    }
+
+    if (req) {
+        return;
+    }
+
+    const auto& rawURL = urlOrTileset.get<std::string>();
+    const auto& url = util::mapbox::canonicalizeSourceURL(fileSource.getResourceOptions().tileServerOptions(), rawURL);
+
+    req = fileSource.request(Resource::source(url), [this, url, &fileSource](const Response& res) {
+        if (res.error) {
+            observer->onSourceError(*this, std::make_exception_ptr(std::runtime_error(res.error->message)));
+        } else if (res.notModified) {
+            return;
+        } else if (res.noContent) {
+            observer->onSourceError(*this, std::make_exception_ptr(std::runtime_error("unexpectedly empty TileJSON")));
+        } else {
+            conversion::Error error;
+            std::optional<Tileset> tileset = conversion::convertJSON<Tileset>(*res.data, error);
+            if (!tileset) {
+                observer->onSourceError(*this, std::make_exception_ptr(util::StyleParseException(error.message)));
+                return;
+            }
+            const auto& tileServerOptions = fileSource.getResourceOptions().tileServerOptions();
+            if (tileServerOptions.uriSchemeAlias() == "mapbox") {
```

Comment:
Previously, this check for mapbox as the uri scheme was only present in `RasterSource`. It was added in #1171 . This change would effectively add this check for `VectorSource` as well. It appears this only has to do with the built-in tile server schemes like maptiler and mapbox. I verified both raster and vector maps from maptiler uris worked as expected. Let me know if there are other implications to this.